perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -518,6 +643,240 @@ def SingleStreamThroughputBenchmark(storage_schema, host_to_connect=None):
 
 
+def RunThreadedWorkers(worker, worker_args, per_thread_args=None):
+  """Run a worker function in many threads, then gather and return the results.
+
+  Args:
+    worker: either WriteWorker or ReadWorker. The worker function to call.
+    worker_args: a tuple. The arguments to pass to the worker function. The
+      result queue and stream number will be appended as the last two arguments.
+    per_thread_args: if given, an array with length equal to the
+      number of threads. Thread number i will be passed
+      per_thread_args[i] after its regular arguments and before the
+      result queue and stream number.
+
+  Returns:
+    A list of the results returned by the workers.
+  """
+
+  result_queue = Queue.Queue()
+
+  logging.info('Creating %s threads', FLAGS.num_streams)
+  if per_thread_args is None:
+    threads = [threading.Thread(target=worker,
+                                args=worker_args + (result_queue, i))
+               for i in xrange(FLAGS.num_streams)]
+  else:
+    threads = [threading.Thread(target=worker,
+                                args=worker_args +
+                                (per_thread_args[i],) +
+                                (result_queue, i))
+               for i in xrange(FLAGS.num_streams)]
+  logging.info('Threads created. Starting threads.')
+  for thread in threads:
+    thread.start()
+  logging.info('Threads started.')
+  while True:
+    time.sleep(THREAD_STATUS_LOG_INTERVAL)
+    num_alive = sum((thread.isAlive() for thread in threads))
+    if num_alive == 0:
+      break
+    else:
+      logging.info('%s of %s threads are still working.',
+                   num_alive, FLAGS.num_streams)
+  logging.info('All threads complete.')
+
+  results = []
+
+  try:
+    while True:
+      res = result_queue.get_nowait()
+      results.append(res)
+  except Queue.Empty:
+    pass
+
+  return results
+
+
+def MultiStreamWrites(storage_schema, host_to_connect=None):
+  """Run multi-stream write threads."""
+
+  size_distribution = yaml.load(cStringIO.StringIO(FLAGS.object_sizes))
+
+  payload_bytes, payload_string = GenerateWritePayload(
+      MaxSizeInDistribution(size_distribution))
+
+  results = RunThreadedWorkers(
+      WriteWorker,
+      (storage_schema,
+       host_to_connect,
+       payload_bytes,
+       payload_string,
+       size_distribution,
+       FLAGS.objects_per_stream,
+       FLAGS.start_time))
+
+  # object_records is the data we leave on the VM for future reads. We
+  # need to pass data to the reader so it will know what object names
+  # it can read. The file format is a JSON list of lists in the format
+  #  [ [object_name_1, object_size_1],
+  #    [object_name_2, object_size_2],
+  #    ... ]
+  if FLAGS.objects_written_file is not None:
+    object_records = []
+    for result in results:
+      for name, size in zip(result['object_names'], result['sizes']):
+        object_records.append([name, size])
+    if os.path.exists(FLAGS.objects_written_file):
```

Comment:
[Not a major issue] 
please consider wrapping these file IO lines with try-except, so if we fail to write to the object names file for any reason, we still get to return the results from our successful benchmarking runs to our caller. but we do need to log a big warning to indicate that file writing has failed, and don't issue read test unless they have an alternative/good object name file.